pkg/cdi/container-edits.go

```diff
@@ -78,12 +83,44 @@ func (e *ContainerEdits) Apply(spec *oci.Spec) error {
 	if len(e.Env) > 0 {
 		specgen.AddMultipleProcessEnv(e.Env)
 	}
+
 	for _, d := range e.DeviceNodes {
-		specgen.AddDevice(d.ToOCI())
+		dev := d.ToOCI()
+		if err := fillMissingInfo(&dev); err != nil {
```

Comment:
Should we always fill the missing information? Are there not use cases where we do not want to query the devices -- for example if the spec is already complete?

Reply:
The called function does nothing if all the mandatory information is already present.